zebra-state/src/sled_state.rs

```diff
@@ -112,59 +124,24 @@ impl FinalizedState {
     /// Flushes sled trees before exiting.
     ///
     /// `called_from` and `block_hash` are used for assertions and logging.
-    fn stop_if_at_height_limit(
-        &self,
-        called_from: StopCheckContext,
-        block_height: block::Height,
-        block_hash: block::Hash,
-    ) {
+    fn is_at_stop_height(&self, block_height: block::Height, block_hash: block::Hash) -> bool {
         let debug_stop_at_height = match self.debug_stop_at_height {
             Some(debug_stop_at_height) => debug_stop_at_height,
-            None => return,
+            None => return false,
         };
 
         if block_height < debug_stop_at_height {
-            return;
-        }
-
-        // this error is expected on load, but unexpected on commit
-        if block_height > debug_stop_at_height {
-            if called_from == StopCheckContext::OnLoad {
-                tracing::error!(
-                    ?debug_stop_at_height,
-                    ?called_from,
-                    ?block_height,
-                    ?block_hash,
-                    "previous state height is greater than the stop height",
-                );
-            } else {
-                unreachable!("committed blocks must be committed in order");
-            }
-        }
-
-        // Don't sync when the trees have just been opened
-        if called_from == StopCheckContext::OnCommit {
-            if let Err(e) = self.flush() {
-                tracing::error!(
-                    ?e,
-                    ?debug_stop_at_height,
-                    ?called_from,
-                    ?block_height,
-                    ?block_hash,
-                    "error flushing sled state before stopping"
-                );
-            }
+            return false;
         }
 
         tracing::info!(
             ?debug_stop_at_height,
-            ?called_from,
             ?block_height,
             ?block_hash,
             "stopping at configured height"
         );
```

Comment:
How can we include the calling function in this info message?
It's really important information if we need to diagnose bugs in this feature.

Should we just duplicate this info message in both callers?

Reply:
`#[track_caller]` TIME, WOO LETS GOOOO

Reply:
oh I see the problem, you want it for the assertion because one of the tests looks for the specific value of CalledFrom

Reply:
I think we should change the log message instead, rather than relying on function names to stay the same.